napari/_qt/widgets/qt_viewer_buttons.py

```diff
@@ -102,6 +102,14 @@ def __init__(self, viewer):
         self.viewer = viewer
         action_manager.context['viewer'] = viewer
 
+        def active_layer():
```

Comment:
should this take `self`, and use `self.viewer` everywhere?

Reply:
Sorry for the delay in responding, some issues with travelling and just starting to get back online today.

`action_manager.context` can either takes objects or callables with no arguments. So this is a closure.
I could change how action_manager's context works.

Reply:
(but yet to self.viewer everywhere)

Reply:
i'm ok with this, just wanted to check! we can leave open in case others have thoughts here too